gammapy/spectrum/energy.py

```diff
@@ -0,0 +1,293 @@
+                raise ValueError("Requested unit {0} is not an"
+                                 " energy unit".format(unit))
+
+        if isinstance(energy, u.Quantity):
+            # also True if energy is of type Energy
+            if unit is not None:
+                energy = energy.to(unit).value
+            else:
+                unit = energy.unit
+                if not unit.is_equivalent(u.eV):
+                    raise ValueError("Given quantity {0} is not an"
+                                     " energy".format(energy))
+                energy = energy.value
+
+        else:
+            if unit is None:
+                raise ValueError("No unit given")
+
+        self = super(Energy, cls).__new__(cls, energy, unit,
+                                          dtype=dtype, copy=copy)
+
+        # Interesting once  energy bounds are stored
+        self._nbins = self.size
+
+        return self
+
+    def __array_finalize__(self, obj):
+        if obj is None:
+            return
+        self._nbins = self.size
+        self._unit = getattr(obj, '_unit', None)
+
+    def __quantity_subclass__(self, unit):
+        unit = u.Unit(unit)
+        if unit.is_equivalent(u.eV):
+            return Energy, True
+        else:
+            return super(Energy, self).__quantity_subclass__(unit)[0], False
+
+    @property
+    def nbins(self):
+        """
+        The number of bins
+        """
+        return self._nbins
+
+    @staticmethod
+    def equal_log_spacing(emin, emax, nbins, unit=None):
+        """Create Energy with equal log-spacing.
+
+        if no unit is given, it will be taken from emax
+
+        Parameters
+        ----------
+        emin : `~astropy.units.Quantity`, float
+            Lowest energy bin
+        emax : `~astropy.units.Quantity`, float
+            Highest energy bin
+        bins : int
+            Number of bins
+        unit : `~astropy.units.UnitBase`, str
+            Energy unit
+
+        Returns
+        -------
+        Energy
+        """
+
+        if unit is None:
+            unit = emax.unit
+
+        emin = Quantity(emin, unit)
+        emax = Quantity(emax, unit)
+
+        x_min, x_max = np.log10([emin.value, emax.value])
+        energy = np.logspace(x_min, x_max, nbins)
+
+        return Energy(energy, unit)
+
+    @staticmethod
+    def from_fits(hdu, unit=None):
+        """Read ENERGIES fits extension.
+
+        Parameters
+        ----------
+        hdu: `~astropy.io.fits.BinTableHDU`
+            ``ENERGIES`` extensions.
+        unit : `~astropy.units.UnitBase`, str
+            Energy unit
+
+        Returns
+        ------
+        Energy
+        """
+
+        header = hdu.header
+        unit = header.get('TUNIT1')
+        if unit is None:
+            raise ValueError("No energy unit could be found in the header of."
+                             "{0}. Please specifiy a unit".format(header.name))
+
+        energy = Quantity(hdu.data['Energy'], unit)
```

Comment:
@cdeil: here it could make sene to use staticmethod, since you like the FITS I/O to be quite explicit, right? That way an ENERGIES extension could only go to the Energy class, whereas an EBOUNDS extension would be restriced to the EnergyBounds class
If we use classmethod here than we could read an ENERGIES extension with both the class Energy and the daugther-class EnergyBounds, correct?